lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+                     dyn_cast<arith::ConstantOp>(val.getDefiningOp())) {
+        // TODO: Add constants
+      } else {
+        mlir::emitError(op.getLoc(), "Operands must be results of GetAccessOp, "
+                                     "LoadOp, TaskletNode or ConstantOp");
+        return failure();
+      }
+    }
+  } else {
+    // calls nested SDFG
+    SDFGNode sdfg = op.getSDFG();
+    for (unsigned i = 0; i < op.getNumOperands(); ++i) {
+      Value val = op.getOperand(i);
+      if (GetAccessOp acc = dyn_cast<GetAccessOp>(val.getDefiningOp())) {
+        if (printAccessSDFGEdge(acc, sdfg, i, jemit).failed())
+          return failure();
+      } else {
+        mlir::emitError(op.getLoc(),
+                        "Operands must be results of GetAccessOp, LoadOp, "
+                        "TaskletNode or ConstantOp");
+        return failure();
+      }
+    }
+  }
+
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// LibCallOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateLibCallToSDFG(LibCallOp &op, JsonEmitter &jemit) {
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// AllocSymbolOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateAllocSymbolToSDFG(AllocSymbolOp &op,
+                                         JsonEmitter &jemit) {
+  jemit.printKVPair(op.sym(), "int64");
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// SymbolExprOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateSymbolExprToSDFG(SymOp &op, JsonEmitter &jemit) {
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// Translate type
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateTypeToSDFG(Type &t, Location loc, JsonEmitter &jemit,
+                                  StringRef key) {
+  if (t.isF64()) {
+    jemit.printKVPair(key, "float64");
+    return success();
+  }
+
+  if (t.isF32()) {
+    jemit.printKVPair(key, "float32");
+    return success();
+  }
+
+  if (t.isInteger(64)) {
+    jemit.printKVPair(key, "int64");
+    return success();
+  }
+
+  if (t.isInteger(32)) {
+    jemit.printKVPair(key, "int32");
+    return success();
+  }
+
+  std::string type;
+  llvm::raw_string_ostream typeStream(type);
+  t.print(typeStream);
+
+  mlir::emitError(loc, "Unsupported type: " + type);
+  return failure();
+}
+
+//===----------------------------------------------------------------------===//
+// Op contains Attr
+//===----------------------------------------------------------------------===//
+
+bool containsAttr(Operation &op, StringRef attrName) {
+  // TODO: Replace with hasAttr
```

Comment:
What's the significance of this comment/TODO?

Reply:
There is already a `hasAttr` operation that does exactly what this helper function does. No need for this helper function. (Removed)